src/main/kotlin/uk/gov/dwp/dataworks/controllers/ConnectionController.kt

```diff
@@ -26,8 +33,21 @@ class ConnectionController {
     ])
     @PostMapping("/connect")
     @ResponseStatus(HttpStatus.OK)
-    fun connect(@RequestHeader("Authorization") token: String) {
-        authService.validate(token)
+    fun connect(@RequestHeader("Authorization") token: String, @RequestBody requestBody: uk.gov.dwp.dataworks.model.Model): String {
+        val jwtObject = authService.validate(token)
+        if (existingUserServiceCheck.check(jwtObject.userName, configService.getStringConfig(ConfigKey.ECS_CLUSTER_NAME))){
+            UserContainerController.logger.info("Redirecting user to running containers, as they exist")
+        } else {
+            taskDeploymentService.taskDefinitionWithOverride(
+                    jwtObject.userName,
+                    requestBody.containerPort,
+                    requestBody.jupyterCpu,
+                    requestBody.jupyterMemory,
+                    requestBody.additionalPermissions
+            )
+            UserContainerController.logger.info("Submitted request", "cluster_name" to configService.getStringConfig(ConfigKey.ECS_CLUSTER_NAME), "user_name" to jwtObject.userName)
+        }
+        return "${configService.getStringConfig(ConfigKey.USER_CONTAINER_URL)}/${jwtObject.userName}"
```

Comment:
I left it as 2 endpoints, as this was requested. I wasn't sure if it was a good idea to combine them, as this would allow for a non-validated call to the connect endpoint.